mlir/lib/IR/PatternMatch.cpp

```diff
@@ -97,6 +98,7 @@ LogicalResult RewritePattern::match(Operation *op) const {
 /// Out-of-line vtable anchor.
 void RewritePattern::anchor() {}
 
+#if MLIR_ENABLE_PDL
```

Comment:
Can we extract the code in this file into a new .cpp and make it its own CMake library to manage enabling/disabling?